### PR TITLE
Upgrading to gradle 7.6.4

### PR DESCRIPTION
**Pull request by jiksa**
Upgraded to gradle 7.6.4 .

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,65 @@
+plugins {
+    id 'java'
```
Comment (sujith-es) [inline build.gradle:2]: Could you add an `application` plugin

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    jcenter() // You can use any other repository you need
+}
+
+dependencies {
+    testImplementation 'junit:junit:4.12'
+    // Add more dependencies as needed
+}
+
+// Define source and test source directories
+sourceSets {
```
Comment (sujith-es) [inline build.gradle]: We are not changing or adding any other folder. Going with the standard Java folder structure.
Do we need this?

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    main {
+        java.srcDirs = ['src/main/java']
+    }
+    test {
+        java.srcDirs = ['src/test/java']
+    }
+}
+
+// Define the 'compile' task
+task compileMain(type: JavaCompile) {
```
Comment (sujith-es) [inline build.gradle]: Is this required? Going with standard Gradle structure.

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    source = [sourceSets.main.java]
+    destinationDirectory = file("${buildDir}/classes/main")
+    classpath = sourceSets.main.runtimeClasspath
+}
+
+// Define the 'testCompile' task
+task testCompile(type: JavaCompile) {
```
Comment (sujith-es) [inline build.gradle]: Required?

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+// Define the 'test' task
+test {
+    testClassesDirs = sourceSets.test.output.classesDirs
+    classpath += sourceSets.main.runtimeClasspath
+}
+
+// Define the 'package' task
+task packageJar(type: Jar) {
```
Comment (sujith-es) [inline build.gradle]: can we use the jar code block here? We have not other requirement to extend it.

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+// Define the 'run' task
+task run(type: JavaExec) {
+    dependsOn packageJar
+    mainClass = 'com.hello.HelloWorld'
+    classpath = files("${buildDir}/libs/${project.name}.jar")
+}
+
+// Define the 'clean' task
+clean {
```
Comment (sujith-es) [inline build.gradle:38]: 👍

File: gradle.properties [hunk 7]
```diff
@@ -0,0 +1 @@
+org.gradle.java.home = /Library/Java/JavaVirtualMachines/jdk1.8.0_202.jdk/Contents/Home
```
Comment (sujith-es) [inline gradle.properties]: what if the other engineer has a different path?

**Review by sujith-es — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (jiksa) [reply build.gradle:2]: added.

File: build.gradle (hunk 2 above)
Comment (jiksa) [reply build.gradle]: removed sourceSet.

File: build.gradle (hunk 3 above)
Comment (jiksa) [reply build.gradle]: removed.

File: build.gradle (hunk 4 above)
Comment (jiksa) [reply build.gradle]: removed

File: build.gradle (hunk 5 above)
Comment (jiksa) [reply build.gradle]: replaced with jar.

File: gradle.properties (hunk 7 above)
Comment (jiksa) [reply gradle.properties]: In my intellij idea settings , I missed to set Gradle JVM , to point to JAVA_HOME. So , I was facing an issue , each time had to run the 'export JAVA_HOME=/path'  command in project root folder . I was using mac os with intellij idea IDE. This is fixed in my local environment and hence I remove gradle.properties which is not required.

**Review by sujith-es — APPROVED** [review]
👍

**Pull request closed without merging**